bindings/pyroot/pythonizations/python/ROOT/_pythonization/_pyz_utils.py

```diff
@@ -0,0 +1,134 @@
+################################################################################
+
+import abc
+import sys
+
+
+class MethodTemplateGetter(object):
+    '''
+    Instances of this class can be injected in class proxies to replace method
+    templates that we want to pythonize. Similarly to what `partialmethod`
+    does, this class implements `__get__` to return a wrapper object of a
+    method that is bound to an instance of the pythonized class. Such object is
+    both callable and subscriptable.
+
+    Attributes:
+        _original_method (cppyy TemplateProxy): original cppyy method template
+            being pythonized.
+        _wrapper_class (subclass of MethodTemplateWrapper): class that wraps a
+            pythonized method template.
+        _extra_args (tuple): extra arguments to be used by the wrapper object
+            when receiving a call.
+    '''
+
+    def __init__(self, original_method, wrapper_class, *extra_args):
+        '''
+        Initializes the getter object of a method template.
+        Saves the original implementation of the method template to be replaced
+        (i.e. the one provided by cppyy) so that it can be later used in the
+        implementation of the pythonization by the wrapper object.
+
+        Args:
+            original_method (cppyy TemplateProxy): original cppyy method
+                template being pythonized.
+            wrapper_class (subclass of MethodTemplateWrapper): class that wraps
+                a pythonized method template.
+            extra_args (tuple, optional): extra arguments to be used by the wrapper
+                object when receiving a call.
+        '''
+        self._original_method = original_method
+        self._wrapper_class = wrapper_class
+        self._extra_args = extra_args
+
+    def __get__(self, instance, instance_type):
+        '''
+        Creates and returns a wrapper object for a method template. The type of
+        the wrapper is a subclass of MethodTemplateWrapper.
+        By implementing `__get__`, we obtain a handle to the instance of the
+        pythonized class on which the application accessed the method template.
+        That allows us to get an original implementation of the method template
+        that is bound to that instance, and pass such implementation along to
+        the wrapper object for later use.
+
+        Args:
+            instance (class instance): instance of the pythonized class on
+                which the application accessed the method template.
+            instance_type (class type): type of the instance.
+
+        Returns:
+            instance of MethodTemplateWrapper subclass: contains a handle to
+                the original implementation of the method template that is
+                bound to `instance` and, possibly, some extra arguments to be
+                used when receiving a call.
+        '''
+        bound_method = self._original_method.__get__(instance, instance_type)
```

Comment:
From the current docs: `By implementing `__get__`, we obtain a handle to the instance of the pythonized class on which the application accessed the method template. That allows us to get an original implementation of the method template that is bound to that instance, and pass such implementation along to the wrapper object for later use.` - I think this already explains that the goal is to get a bound method.
Also I'd like to keep `original` in the name because I want to convey that it is the original (cppyy) implementation. The fact that it becomes `bound` is then explained by the `bound_method` variable. `partialmethod` can't be used here because the result of it would be something that I cannot subscript afterwards - that's why keeping the cppyy template proxy is important.

Reply:
that comment tells me why we need `MethodTemplateGetter` rather than just assigning `MethodTemplateWrapper` to the attribute. indirectly it also hints at what _this_ `__get__` here does, but at least for me it took a while to realize that it transforms an unbound method to a bound method. 

> partialmethod can't be used here because the result of it would be something that I cannot subscript afterwards

ah, didn't realize that. this would also be important to mention in a comment here.